web/source/osk/visualKeyboard.ts

```diff
@@ -59,14 +58,13 @@ namespace com.keyman.osk {
     touchCount: number;
     currentTarget: KeyElement;
 
-    // Popup key management
-    popupBaseKey: KeyElement;
-    popupPending: boolean = false;
-    subkeyDelayTimer: number;
-    popupDelay: number = 500;
+    // Used by embedded-mode's globe key
     menuEvent: KeyElement; // Used by embedded-mode.
+
+    // Popup key management
     keytip: KeyTip;
-    subkeyPopup: browser.SubkeyPopup;
+    pendingSubkey: PendingGesture;
+    subkeyGesture: RealizedGesture;
```

Comment:
All longpress & subkey state information is now encapsulated by these two references and the objects they represent.  Yay, encapsulation!